### PR TITLE
Bag quantity desktop/mobile view fix

### PR DESCRIPTION
**Pull request by ShavingSeagull**
### Problem overview
The bag was designed with a desktop and mobile view in place, differentiated on-screen via the hiding of one or the other. This posed an issue with the quantity inputs, as these were using IDs that the JS picked up on in the **quantity_input_script.html**. Having two different versions using the same IDs on the same page obviously caused an issue - while one version is always hidden, it is still rendered in the DOM.

### Fix overview
The fix only targets three files:
- quantity-form.html (in the 'bag' app)
- quantity_input_script.html (in the 'includes' directory of the 'products' app)
- product_detail.html (also in the 'products' app)

Chris designed Boutique to use DOM traversal. This is likely due to the same quantity_input_script being used on the product_detail page as well as the bag. The fix has used the same approach to keep continuity. Primarily, the major alterations have been to shift from using IDs to classes in the two versions of the view. This, of course, required some tweaking in the JS as well. The desktop view previously didn't have its quantity adjustment buttons greyed out when the quantity hit the goal posts to disable them; the quantity itself also didn't carry over from one version of the view to the other - change the quantity of an item to 3 on the mobile view and it would still be 1 on the desktop version.

These values now carry across to the two different versions. The quantity buttons now grey-out correctly for both versions. There was an unforeseen bug with this fix surrounding product sizes - if the same item was in the cart twice with differing sizes, my fix would change the quantities for all of them.

The fix for this was to use the size alongside the item's ID in order to only affect all the quantity fields of that particular item and not others with the same ID, but a different size. The fix utilises some additional data attributes, as well as some new classes.

### TL;DR
The changes primmarily add some new classes and data attributes to the HTML files. There are also a few alterations to accommodate this in the JS, though these aren't major. No new functions have been created; all changes are made to existing ones, with a new parameter being added in places. The product_detail.html also needed changing as it uses the same quantity script as the bag - fix the bag and you'll break the product page, hence why the fix needed carrying over to both.

### CODE REVIEW

**Comment by ShavingSeagull** [comment]
Sorry, Chris, thought I was on the CI forked version of the Boutique project *facepalm*
